### PR TITLE
refactor(db access) - Extract direct db access from fxa-graphql-api to fxa-shared

### PR DESCRIPTION
**Pull request by jackiemunroe**
## Because:

- We want to get away from more stored procedures

## This commit:

- Extracts direct db access out of fxa-graphql-api and moved it to fxa-shared

## Issue that this pull request solves

Closes: # 6338

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.

### CODE REVIEW

File: packages/fxa-shared/package.json [hunk 1]
```diff
@@ -36,19 +36,24 @@
...
+    "@types/node": "^14.11.1",
     "@types/proxyquire": "^1.3.28",
     "@types/redis": "^2.8.16",
     "@types/sinon": "^9.0.4",
     "audit-filter": "^0.5.0",
     "chai": "^4.2.0",
+    "chance": "^1.1.7",
     "eslint": "^6.8.0",
     "eslint-plugin-fxa": "^2.0.2",
+    "fs": "^0.0.1-security",
```
Comment (dannycoates) [inline packages/fxa-shared/package.json:53]: ???

File: packages/fxa-shared/package.json [hunk 2]
```diff
@@ -36,19 +36,24 @@
   "homepage": "https://github.com/mozilla/fxa/tree/main/packages/fxa-shared#readme",
   "devDependencies": {
     "@types/chai": "^4.2.11",
+    "@types/chance": "^1",
     "@types/generic-pool": "^3.1.9",
     "@types/lodash.omitby": "^4",
     "@types/lodash.pick": "^4",
     "@types/mocha": "^7.0.2",
+    "@types/node": "^14.11.1",
```
Comment (dannycoates) [inline packages/fxa-shared/package.json:44]: this should be `^12.12.38` until we're using node 14

**Comment by jackiemunroe** [comment]
@dannycoates i feel like i am super close, but still getting some failing tests. i'm not sure if it's due to dependencies. would you mind taking a look?

**Review by bbangert — APPROVED** [review]
LGTM! I think it'll need a rebase, as @dannycoates had mentioned to me before, just running `yarn` after the rebase halts can fix the `yarn.lock` file on its own typically.

File: packages/fxa-graphql-api/src/config.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ import path from 'path';
 convict.addFormats(require('convict-format-with-moment'));
 convict.addFormats(require('convict-format-with-validator'));
 
+/** @depricated */
```
[the remark this answers is not in the document]
Comment (jackiemunroe) [reply packages/fxa-graphql-api/src/config.ts]: updated. i had flagged the code as depricated, so i could keep track of it during the migration process.

**Pull request merged**